### PR TITLE
Patch slurm 17 11 11

### PR DESCRIPTION
**Pull request by diedpigs**
Since Open HPC 1.3.6 updated slurm to 17.11.10 that causes environment not passing into slurm job, we need to install/update slurm to 17.11.11 manually. This patch added a separated ansible task to install slurm 17.11.11 which i build via modifying source rpm provided by ohpc 1.3.6. The task should be removed afteer ohpc update to newer version of slurm.

### CODE REVIEW

**Comment by jprorama** [comment]
How are we handling the kernel mismatch issue currently?  I can't seem to get past the kernel mismatch error when building from the ground up on ravi89/centos7.5 and I can't get past nmcli errors on eth0 from the stock upstream centos7
```
    ohpc: TASK [pre_ohpc : add public interface to public zone via nmcli] ****************
    ohpc: fatal: [ohpc]: FAILED! => {"changed": true, "cmd": ["nmcli", "connection", "modify", "eth0", "connection.zone", "public"], "delta": "0:00:00.016259", "end": "2018-12-28 13:03:47.014388", "msg": "non-zero return code", "rc": 10, "start": "2018-12-28 13:03:46.998129", "stderr": "Error: unknown connection 'eth0'.", "stderr_lines": ["Error: unknown connection 'eth0'."], "stdout": "", "stdout_lines": []}
```

**Comment by diedpigs** [comment]
with ravi89/centos7.5, i believe it's using predictable interface naming. That's why you got 'unknown connection eth0'. 
About kernel mismatch issue, i think the latest version of centos/7 is now at 3.10.0-957.1.3.el7, but ravi's still at older version.

File: roles/ohpc_slurm_17_11_11/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    yum:
+      name:
+        - "{{ slurm_17_11_11_dest }}/slurm17.11.11/slurm-devel-ohpc-17.11.11-6.2.ohpc.1.3.6.x86_64.rpm"
+        - "{{ slurm_17_11_11_dest }}/slurm17.11.11/slurm-example-configs-ohpc-17.11.11-6.2.ohpc.1.3.6.x86_64.rpm"
+        - "{{ slurm_17_11_11_dest }}/slurm17.11.11/slurm-ohpc-17.11.11-6.2.ohpc.1.3.6.x86_64.rpm"
+        - "{{ slurm_17_11_11_dest }}/slurm17.11.11/slurm-perlapi-ohpc-17.11.11-6.2.ohpc.1.3.6.x86_64.rpm"
+        - "{{ slurm_17_11_11_dest }}/slurm17.11.11/slurm-slurmctld-ohpc-17.11.11-6.2.ohpc.1.3.6.x86_64.rpm"
+        - "{{ slurm_17_11_11_dest }}/slurm17.11.11/slurm-slurmdbd-ohpc-17.11.11-6.2.ohpc.1.3.6.x86_64.rpm"
+
+  - name: Update slurm related packages on compute node
```
Comment (jprorama) [inline roles/ohpc_slurm_17_11_11/tasks/main.yml]: Change the  text to read "...for compute node image" .  We aren't changing this directly on a running compute node but rather maintaining the image that boots the compute node.

**Comment by jprorama** [comment]
Tested this patch in a new environment and it successfully updates the slurm version to 17.11.11.  One thing to note is that the install has to be done before the ood mounts it's dirs from the ohpc master.  The slurm-example-configs-ohpc rpm writes to /opt/ohpc which will be read-only nfs mounted in normal operations.  This shouldn't be an issue during provisioning since the `mount -a` doesn't happen till the final step, so will ignore this minor issue for the PR.  BTW, this is the same approach ohpc master takes when building the compute node images.    In the long run, this slurm example rpm can probably be skipped on the ood node.

**Pull request merged**